lib/expand-ports.js

```diff
@@ -25,12 +23,24 @@ const expandPorts = ports => {
       throw new Error(typeof width);
     }
 
+    return {
+      direction: direction,
+      width: width
+    };
+  }
+};
+
+const expandPorts = ports => {
+  if (ports.constructor !== Object) {
+    return ports.foreach((port) => {
+      port.wire = expandWire(port['wire']);
```

Comment:
Nit: You should be consistent and either write `port.wire` in both places or `port['wire']` in both places, because they are both equivalent syntaxes for accessing the same property. Generally I believe people prefer dot notation for any keys that are statically known.